### PR TITLE
Add billing  UUID to PO

### LINKED ISSUE #26: Add bill to/ship to address to PO
The bill to/ ship to address should be added to the PO. The address is "LTS Acquisitions" the UUID can be found via

https://okapi-cornell-training.folio.ebsco.com/configurations/entries?limit=500&query=(module=TENANT and configName=tenant.addresses)

This is a pretty static thing, I would be comfortable putting it in the env as string or UUID, whatever seems best. Both bill to and ship to are the same.

### PR DESCRIPTION
**Pull request by fereira**
- adds a map  containing billing address as a lookup table
- uses billTo environment variable as a key to get billing address UUID
- validates that billTo and other required env vars have been set

Resolves #26

### CODE REVIEW

File: src/main/java/org/olf/folio/order/listeners/ConfigurationListener.java [hunk 1]
```diff
@@ -27,10 +27,11 @@ public void contextInitialized(ServletContextEvent sce) {
 		 
 				config.setProperty("permELocation", System.getenv("permELocation")); 
 				config.setProperty("fiscalYearCode", System.getenv("fiscalYearCode"));
-				config.setProperty("loanType", System.getenv("loanType"));
-				config.setProperty("textForElectronicResources", System.getenv("textForElectronicResources"));
+				// config.setProperty("loanType", System.getenv("loanType"));
+				// config.setProperty("textForElectronicResources", System.getenv("textForElectronicResources"));
 				config.setProperty("noteType", System.getenv("noteType"));
 				config.setProperty("materialType", System.getenv("materialType"));
+				config.setProperty("billTo", System.getenv("billTo"));
```
Comment (cappadona) [inline src/main/java/org/olf/folio/order/listeners/ConfigurationListener.java:34]: We should add `billTo` to [.env.example](https://github.com/cul-it/order-import-poc/blob/add-billTo/.env.example) and document it in [Docker.md](https://github.com/cul-it/order-import-poc/blob/add-billTo/Docker.md).

File: src/main/java/org/olf/folio/order/util/LookupUtil.java [hunk 2]
```diff
@@ -6,110 +6,207 @@
...
+        
+        JSONArray configsArray = jsonObject.getJSONArray("configs");
+        Iterator elementsIterator = configsArray.iterator();
+        while (elementsIterator.hasNext()) {
+            JSONObject element = (JSONObject) elementsIterator.next();
+             
+            String value = element.getString("value");
+            String uuid = element.getString("id");
+            String configName = element.getString("configName");
+            if (StringUtils.equals(configName, "tenant.addresses")) {
```
Comment (cappadona) [inline src/main/java/org/olf/folio/order/util/LookupUtil.java:188]: If we add the the `query` URL param suggested above, that would ensure only config entries of type `tenant.addresses` would be returned in the API response and then we can eliminate this test.

File: src/main/java/org/olf/folio/order/OrderImport.java [hunk 3]
```diff
@@ -105,12 +112,20 @@ public  JSONArray  upload(String fileName) throws IOException, InterruptedExcept
 			 lookupUtil.setApiService(apiService);
 			 lookupUtil.load();
 			 this.lookupTable = lookupUtil.getReferenceValues(token);
+			 String billingEndpoint = baseOkapEndpoint+"configurations/entries?limit=100";
```
Comment (cappadona) [inline src/main/java/org/olf/folio/order/OrderImport.java:115]: What do you think about dropping the limit and instead using the [`query` URL param](https://s3.amazonaws.com/foliodocs/api/mod-configuration/p/config.html#configurations_entries_get) to only return `tenant.addresses`?

```suggestion
			 String billingEndpoint = baseOkapEndpoint+"configurations/query=(configName==tenant.addresses)";
```

Not only does it simplify the logic for `lookupUtil.getBillingAddresses()`, but it also avoids potentially missing addresses as the number of configuration entries grow beyond some arbitrary limit.

> Configuration entries count currently `56` for training and `58` for test

**Review by cappadona — CHANGES_REQUESTED** [review]
As is, this will not add a billing address to POs when deployed. We need to add the `billTo` environment variable to the `.env` file that is currently [stored in an S3 bucket](https://github.com/cul-it/terraform-cul-ecs-order-import-poc/blob/main/services.auto.tfvars#L12). I will follow up in Slack. 

I also have a couple suggestions for refining the API call when retrieving the tenant addresses and the ensuing logic when building the lookup map.

File: src/main/java/org/olf/folio/order/OrderImport.java (hunk 3 above)
Comment (fereira) [reply src/main/java/org/olf/folio/order/OrderImport.java:115]: The URL you suggested won't work.  This one will:
String endpoint = this.getBaseOkapEndpoint()+"configurations/entries?query=(configName==tenant.addresses)";

File: src/main/java/org/olf/folio/order/listeners/ConfigurationListener.java (hunk 1 above)
Comment (fereira) [reply src/main/java/org/olf/folio/order/listeners/ConfigurationListener.java:34]: I don't know how to or if I can modify the .env file stored in S3.  The Docker.md file doesn't explicitly mention all the env files and just has: "Edit other properties as needed"...is that not sufficient?

File: src/main/java/org/olf/folio/order/util/LookupUtil.java (hunk 2 above)
Comment (fereira) [reply src/main/java/org/olf/folio/order/util/LookupUtil.java:188]: I'll push a change that uses the query parameter.  *all* of the other query endpoints are using a limit though. I've thought about refactoring the LookupUtil class to that it's more explicit in it's queries and how it implements the lookkup table, but that'll take a bit of work for minimal gain.

File: src/main/java/org/olf/folio/order/OrderImport.java (hunk 3 above)
Comment (cappadona) [reply src/main/java/org/olf/folio/order/OrderImport.java]: Sorry for my typo

File: .env.example [hunk 4]
```diff
@@ -9,3 +9,5 @@ loanType=Can circulate
 textForElectronicResources=Available to Snapshot Users
 noteType=General note
 materialType=Book
+buildDir=.
```
Comment (cappadona) [inline .env.example:12]: Let's remove this for now and keep it part of #35 
```suggestion
```

File: src/main/java/org/olf/folio/order/listeners/ConfigurationListener.java (hunk 1 above)
Comment (cappadona) [reply src/main/java/org/olf/folio/order/listeners/ConfigurationListener.java:34]: That's probably sufficient for now, especially if we're providing a usable example value for Cornell. This highlights that we need to revisit overall documentation (README/Docker/Deployment) down the road.

**Review by cappadona — APPROVED** [review]

**Comment by cappadona** [comment]
@fereira I've updated the `.env` in the S3 bucket so feel free to merge at your leisure.

**Pull request merged**